### PR TITLE
unify application database bean and test database bean in selenium #251 remove postgres setting #253

### LINKED ISSUE #251: Is it better to define the connection destination provided by the application to be tested in selenium.properties like selenium.serverUrl?
## Description
Information on the application to be tested is written at [secure-login-infra.properties](https://github.com/terasolunaorg/tutorial-apps-macchinetta-thymeleaf/blob/master/secure-login-demo/src/secure-login-selenium/src/test/resources/META-INF/spring/secure-login-infra.properties) file exist in the selenium project now .

Is it better setting ?

## Possible Solutions
(Please write ideas or candidates of solutions for the problem if you have)

## Affects Version/s
* 5.X.X.RELEASE
* 1.X.X.RELEASE

## Fix Version/s
- [x] 5.5.0 (master)
- [x] 5.4.2 (5.4.x)
- [x] 5.3.3 (5.3.x)
- [x] 5.2.2 (5.2.x)
- [x] 5.1.2 (5.1.x)
- [x] 1.6.0 (m-j/master)
- [x] 1.6.0 (m-t/master)

## Final Solution
(To be written later by project member)

## Issue Links
* #XXX

### PR DESCRIPTION
**Pull request by btyanagawamg**
Please review #225 #251 #253 .

I fix bug for #225 correspondence at 84bf6696729e2ec5d4bea57b8c9294afc694e297 commit, and fix #251 Issue at d616c6308f466824129f7ac2b7136365575c3ef2 commit.

unify seleniumContext.xml datasource setting

### CODE REVIEW

**Comment by yoshikawaa** [comment]
@btyanagawamg See https://github.com/terasolunaorg/tutorial-apps/issues/251#issuecomment-370993827 and correct PR title.

File: secure-login-demo/scripts/convert-secure-login-test.sh [hunk 1]
```diff
@@ -86,22 +87,35 @@ selenium.restOperations.connectTimeout=-1
...
+selenium.target.application.database=H2
+selenium.target.application.database.url=jdbc:h2:tcp://'"${HOST_IP}"':'"${APSRV_H2DB_PORT}"'/mem:secure-login
+selenium.target.application.database.username=sa
+selenium.target.application.database.password=
+selenium.target.application.database.driverClassName=org.h2.Driver
+# connection pool
+selenium.cp.maxActive=96
+selenium.cp.maxIdle=16
+selenium.cp.minIdle=0
+selenium.cp.maxWait=60000
```
Comment (yoshikawaa) [inline secure-login-demo/scripts/convert-secure-login-test.sh]: - [ ] I think that it is better to make this the same definition style as https://github.com/terasolunaorg/terasoluna-gfw-functionaltest/blob/2f333deb382e0772b0c51ffc00601f303050b93c/terasoluna-gfw-functionaltest-selenium/src/test/resources/META-INF/spring/seleniumContext.xml#L10-L19.

Comment (btyanagawamg) [reply secure-login-demo/scripts/convert-secure-login-test.sh]: The setting of the proposed place is the setting of the connection destination of db that saves the test log .
"selenium/-infra.xml" has settings used to connect to db used by the application, not test .
However, it is not necessary for the selenium project to have the DB settings used by the application .
So, at the above commit, I changed the setting of the DB connection defined in the env project to also be used in the selenium project .

**Comment by btyanagawamg** [comment]
Fix sed bug
71b5651613817144fbe54c2cc4d95912ad401de3
see https://github.com/terasolunaorg/tutorial-apps/blob/release/5.4.1.RELEASE/secure-login-demo/secure-login-demo/secure-login-web/src/main/webapp/WEB-INF/web.xml#L7

File: secure-login-demo/scripts/convert-secure-login-test.sh [hunk 2]
```diff
@@ -94,7 +102,7 @@ sed -i -e 's|</beans>|\
     <bean id="realDataSource" class="org.apache.commons.dbcp2.BasicDataSource"\
         destroy-method="close">\
         <property name="driverClassName" value="${database.driverClassName}" />\
-        <property name="url" value="${database.url}" />\
+        <property name="url" value="jdbc:h2:tcp://${database.host}:${database.port}/mem:secure-login" />\
```
Comment (yoshikawaa) [inline secure-login-demo/scripts/convert-secure-login-test.sh]: - [ ] Please unite h2 databases (`mem:secure-login`, `mem:secure-login-test`) as well as terasoluna-gfw-functionaltest.

Comment (btyanagawamg) [reply secure-login-demo/scripts/convert-secure-login-test.sh]: 9db39cb07b1ae72bf04972c66b43824dfacb06f9 
I unified the names of h2 database .

File: secure-login-demo/scripts/convert-secure-login-test.sh [hunk 3]
```diff
@@ -100,37 +99,24 @@ selenium.dbResetScript=database/H2-datareload.sql' >> $i ;done
...
         <property name="driverClassName" value="org.h2.Driver" />\
-        <property name="url" value="jdbc:h2:tcp://${selenium.logDbHost}:${selenium.logDbPort}/mem:secure-login" />\
+        <property name="url" value="jdbc:h2:tcp://${selenium.dbHost}:${selenium.dbPort}/mem:secure-login" />\
         <property name="username" value="sa" />\
         <property name="password" value="" />\
         <property name="defaultAutoCommit" value="false" />\
+        <property name="maxTotal" value="${selenium.cp.maxActive}" />\
+        <property name="maxIdle" value="${selenium.cp.maxIdle}" />\
+        <property name="minIdle" value="${selenium.cp.minIdle}" />\
+        <property name="maxWaitMillis" value="${selenium.cp.maxWait}" />\
```
Comment (yoshikawaa) [inline secure-login-demo/scripts/convert-secure-login-test.sh]: - [ ] Are these necessary? (maxTotal, maxIdle, minIdle, maxWaitMillis)

Comment (btyanagawamg) [reply secure-login-demo/scripts/convert-secure-login-test.sh]: 0c55e28e7eaefd206ce461fa0937942f5a8f6ddc
I tried above commit and got selenium error .
I will do it at another issue .

File: secure-login-demo/src/secure-login-selenium/src/test/java/com/example/securelogin/selenium/loginform/AccountLockTest.java [hunk 4]
```diff
@@ -55,7 +56,15 @@
...
-        populator.populate(dataSource.getConnection());
+        Connection connection = null;
+        try {
+            connection = dataSource.getConnection();
+            populator.populate(connection);
+        } finally {
+            if (connection != null) {
+                connection.close();
+            }
+        }
```
Comment (yoshikawaa) [inline secure-login-demo/src/secure-login-selenium/src/test/java/com/example/securelogin/selenium/loginform/AccountLockTest.java]: - [ ] Please re-write using try-with-resources style.

**Pull request merged**